### PR TITLE
Added missing api methods for the integrations installation api

### LINKED ISSUE #516: Add "Find installations" wrapper to Integrations
There is an API call you can make to fetch your installations. 

`GET /integration/installations`
Referenced here: https://developer.github.com/v3/integrations/

This can easily be added to Api/Integrations.php by adding these lines:

```
public function findInstallations()
    {
        return $this->get('/integration/installations');
    }
```
I realize that this can be accomplished using the generic `$github->getHttpClient()->get('integration/installations')`, but this convenience method would be nice.

### PR DESCRIPTION
**Pull request by acrobat**
Missing methds for the integrations installation api. See https://developer.github.com/v3/integrations/installations/#installations

And added the [`findInstallations` api method](https://developer.github.com/v3/integrations/#find-installations). Closes #516

### CODE REVIEW

File: lib/Github/Api/Integrations.php [hunk 1]
```diff
@@ -26,4 +26,52 @@ public function createInstallationToken($installationId, $userId = null)
 
         return $this->post('/installations/'.rawurlencode($installationId).'/access_tokens', $parameters);
     }
+
+    /**
+     * Find all installations for the authenticated integration.
+     *
```
Comment (Nyholm) [inline lib/Github/Api/Integrations.php]: Add `@link https://developer.github.com/v3/integrations/#find-installations`

File: lib/Github/Api/Integrations.php [hunk 2]
```diff
@@ -26,4 +26,52 @@ public function createInstallationToken($installationId, $userId = null)
...
+    }
+
+    /**
+     * List repositories that are accessible to the authenticated installation.
+     *
+     * @param int $userId
+     *
+     * @return array
+     */
+    public function listRepositories($userId)
```
Comment (Nyholm) [inline lib/Github/Api/Integrations.php]: $userId should be optional, right?

File: lib/Github/Api/Integrations.php [hunk 3]
```diff
@@ -26,4 +26,52 @@ public function createInstallationToken($installationId, $userId = null)
...
+     * @return array
+     */
+    public function findInstallations()
+    {
+        return $this->get('/integration/installations');
+    }
+
+    /**
+     * List repositories that are accessible to the authenticated installation.
+     *
```
Comment (Nyholm) [inline lib/Github/Api/Integrations.php]: Add link to https://developer.github.com/v3/integrations/installations/#list-repositories

File: lib/Github/Api/Integrations.php [hunk 4]
```diff
@@ -26,4 +26,52 @@ public function createInstallationToken($installationId, $userId = null)
...
+     * @return array
+     */
+    public function listRepositories($userId)
+    {
+        return $this->get('/installation/repositories', ['user_id' => $userId]);
+    }
+
+    /**
+     * Add a single repository to an installation.
+     *
```
Comment (Nyholm) [inline lib/Github/Api/Integrations.php]: https://developer.github.com/v3/integrations/installations/#add-repository-to-installation

File: lib/Github/Api/Integrations.php [hunk 5]
```diff
@@ -26,4 +26,52 @@ public function createInstallationToken($installationId, $userId = null)
...
+     * @return array
+     */
+    public function addRepository($installationId, $repositoryId)
+    {
+        return $this->put('/installations/'.rawurlencode($installationId).'/repositories/'.rawurlencode($repositoryId));
+    }
+
+    /**
+     * Remove a single repository from an installation.
+     *
```
Comment (Nyholm) [inline lib/Github/Api/Integrations.php]: https://developer.github.com/v3/integrations/installations/#remove-repository-from-installation

File: test/Github/Tests/Api/IntegrationTest.php [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+     * @test
+     */
+    public function shouldFindRepositoriesForIntegration()
+    {
+        $api = $this->getApiMock();
+        $api->expects($this->once())
+            ->method('get')
+            ->with('/integration/installations');
+
+        $api->findInstallations();
```
Comment (Nyholm) [inline test/Github/Tests/Api/IntegrationTest.php]: Test return value as well.

https://github.com/KnpLabs/php-github-api/blob/master/doc/testing.md

File: test/Github/Tests/Api/IntegrationTest.php [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+     * @test
+     */
+    public function shouldGetRepositoriesFromInstallation()
+    {
+        $api = $this->getApiMock();
+        $api->expects($this->once())
+            ->method('get')
+            ->with('/installation/repositories');
+
+        $api->listRepositories('1234');
```
Comment (Nyholm) [inline test/Github/Tests/Api/IntegrationTest.php]: Test that the parameter is used. Also make a second test where you not using any $userId

File: lib/Github/Api/Integrations.php (hunk 2 above)
Comment (acrobat) [inline lib/Github/Api/Integrations.php]: No if you call the api with the userId you just get a 403 error irc. The repositories are also always cehcked for the current installation and the repo's the user has access to. I guess github will change this api route in the future so the userId is part of the uri.

**Comment by acrobat** [comment]
I've added the links to the docsblocks. I will extend the tests tonight!

File: test/Github/Tests/Api/IntegrationTest.php (hunk 7 above)
Comment (acrobat) [inline test/Github/Tests/Api/IntegrationTest.php]: `$userId`is required so this test should do. But can you give me some example or info on how to test the parameters content? I've thought about this test but I couldn't find on how to retrieve the passed parameters and check if `$params['user_id']` was equal to the passed value

Comment (Nyholm) [inline test/Github/Tests/Api/IntegrationTest.php]: Try something like:

```php
$api->expects($this->once())
->method('get')
->with('/installation/repositories', ['user_id'=>'1234']);
```

File: lib/Github/Api/Integrations.php (hunk 2 above)
Comment (Nyholm) [inline lib/Github/Api/Integrations.php]: Okey. Good. I guess I misunderstood the docs.

**Comment by acrobat** [comment]
@Nyholm I've extended the tests and other comments should be fixed!

**Comment by acrobat** [comment]
@Nyholm any more work needed here?

**Comment by acrobat** [comment]
@Nyholm any more work needed here?

**Comment by Nyholm** [comment]
Sorry for the delay. I will review this PR shortly

**Comment by acrobat** [comment]
Thanks a lot @Nyholm!

**Pull request merged**